Core/Core/Inbox/View/AttachmentsView.swift

```diff
@@ -0,0 +1,92 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+public struct AttachmentsView: View {
+    private let attachments: [File]
+    private let didSelectAttachment: ((File) -> Void)?
+    @State private var isAttachmentDeleted = false
+
+    public init(attachments: [File], didSelectAttachment: ((File) -> Void)? = nil) {
+        self.attachments = attachments
+        self.didSelectAttachment = didSelectAttachment
+    }
+
+    public var body: some View {
+        ScrollView(.horizontal) {
+            HStack(spacing: 8) {
+                ForEach(attachments, id: \.self) { file in
+                    attachmentView(for: file)
+                }
+            }
+        }
+    }
+
+    private func attachmentView(for file: File) -> some View {
```

Comment:
Would it make sense to make an `AttachmentView` the component itself and put the list on the screen/view that displays the attachments? That way an attachment could be used by itself.

Reply:
There was some misunderstanding about this, I proposed that single cards should be their own structs
But I'm fine either way since we are sure it makes no sense to display attachments one by one.